v2/pkg/engine/resolve/errors.go

```diff
@@ -0,0 +1,109 @@
+package resolve
+
+import (
+	"fmt"
+	"strings"
+)
+
+type GraphQLError struct {
+	Message    string          `json:"message"`
+	Locations  []Location      `json:"locations,omitempty"`
+	Path       []string        `json:"path"`
+	Extensions *ErrorExtension `json:"extensions,omitempty"`
+}
+
+type ErrorExtension struct {
+	Code string `json:"code"`
+}
+
+type Location struct {
+	Line   uint32 `json:"line"`
+	Column uint32 `json:"column"`
+}
+
+type SubgraphError struct {
+	SubgraphName string
+	Path         string
+	Reason       string
+	ResponseCode int
+
+	DownstreamErrors []*GraphQLError
+}
+
+func NewSubgraphError(subgraphName, path, reason string, responseCode int) *SubgraphError {
+	return &SubgraphError{
+		SubgraphName: subgraphName,
+		Path:         path,
+		Reason:       reason,
+		ResponseCode: responseCode,
+	}
+}
+
+func (e *SubgraphError) AppendDownstreamError(error *GraphQLError) {
+	e.DownstreamErrors = append(e.DownstreamErrors, error)
+}
+
+func (e *SubgraphError) Error() string {
+
+	var sb strings.Builder
+
+	if e.SubgraphName == "" {
+		sb.WriteString(fmt.Sprintf("Failed to fetch Subgraph at path: '%s'", e.Path))
```

Comment:
Can you match the wording with the existing errors?
https://github.com/wundergraph/cosmo/blob/8aa618e478fee433c6fc4d8311205b9a234bddb5/router-tests/integration_test.go#L765